src/test/java/teammates/logic/core/AccountsLogicTest.java

```diff
@@ -424,4 +427,53 @@ public void testDeleteAccountCascade_nonExistentAccount_shouldPass() {
         assertNotNull(instructorsLogic.getInstructorForEmail(
                 instructor1OfCourse1.getCourseId(), instructor1OfCourse1.getEmail()));
     }
+
+    @Test
+    public void testUpdateReadNotifications() throws Exception {
+        AccountAttributes instructor2OfCourse1 = dataBundle.accounts.get("instructor2OfCourse1");
+        NotificationAttributes notificationAttributes = dataBundle.notifications.get("notification4");
+
+        ______TS("success: mark notification as read and remove expired ones from read status");
+
+        AccountAttributes accountAttributes = accountsLogic.updateReadNotifications(
+                instructor2OfCourse1.getGoogleId(),
+                notificationAttributes.getNotificationId(),
+                notificationAttributes.getEndTime());
+        Map<String, Instant> updatedReadNotifications = accountAttributes.getReadNotifications();
+
+        assertNotNull(updatedReadNotifications.get("notification4"));
+        assertEquals(notificationAttributes.getEndTime(), updatedReadNotifications.get("notification4"));
+
+        for (Map.Entry<String, Instant> notification : updatedReadNotifications.entrySet()) {
+            assertTrue(notification.getValue().isAfter(Instant.now()));
+        }
+
+        ______TS("failure: invalid googleId");
+
+        assertThrows(EntityDoesNotExistException.class,
+                () -> accountsLogic.updateReadNotifications(
+                        "not_exist",
+                        notificationAttributes.getNotificationId(),
+                        notificationAttributes.getEndTime()));
+
+        ______TS("failure: invalid notificationId");
+
+        assertThrows(EntityDoesNotExistException.class,
+                () -> accountsLogic.updateReadNotifications(
+                        instructor2OfCourse1.getGoogleId(),
+                        "invalid_notification_id",
+                        notificationAttributes.getEndTime()));
+
+        ______TS("failure: expired notification");
```

Comment:
Sorry this comment was accidentally removed from my last review, just a small request to group these under a unified logical section, maybe  `______TS("failure: update read notifications with invalid parameter");`

Reply:
sure, i will then remove invidual `______TS` for each parameter and use comments instead